File: src/dataset/cpp/luhn_exercism-Richard-m-j/openAI/ChatGPT4_luhn_v1.cpp
#include "luhn.h"
#include <string>
#include <cctype>
namespace luhn
{
    bool valid(std::string inputNumber)
    {
        int sum = 0;
        int digitCount = 0;
        bool doubleDigit = false;

        for (auto it = inputNumber.rbegin(); it != inputNumber.rend(); ++it)
        {
            if (*it == ' ')
                continue;

            if (!isdigit(*it))
                return false;

            int digit = *it - '0';
            if (doubleDigit)
            {
                digit *= 2;
                if (digit > 9)
                    digit -= 9;
            }

            sum += digit;
            doubleDigit = !doubleDigit;
            ++digitCount;
        }

        return (sum % 10 == 0 && digitCount > 1);
    }
}

File: src/dataset/python/luhn_exercism-fruit-in/openAI/ChatGPT4_luhn_v1.py
class Luhn:
    def __init__(self, card_num):
        card_num = card_num.replace(' ', '')
        self.checksum = 0

        if card_num.isdigit() and len(card_num) > 1:
            self.checksum = sum(
                int(d) if i % 2 == 0 else (2 * int(d) - 9 if 2 * int(d) > 9 else 2 * int(d))
                for i, d in enumerate(reversed(card_num))
            )

    def valid(self):
        return self.checksum % 10 == 0